lib/engine/game/base.rb

```diff
@@ -617,7 +617,7 @@ def stock_round
       end
 
       def new_operating_round(round_num = 1)
-        @log << "-- Operating Round #{@turn}.#{round_num} --"
+        @log << "-- Operating Round #{@turn}.#{round_num} (of #{@operating_rounds}) --"
```

Comment:
why is this duplicated with log_new_round?

looks like i implemented log_new_round but never call it, can you clean this stuff up and just use the one in the round?

Reply:
The easiest thing to do would be to just get rid of `log_new_round`. I spent 5 minutes trying to use `log_new_round` in a nice abstract way, but it's annoying, because by the time you have a `Round` object to call it on, you've already called the `Round` constructor, which does things like have privates pay out, so the round-change message shows up in the log late. We could call `log_new_round` in the constructors of the individual `Round` types, but then we've lost the benefit of writing it just in one place. (We could also separate out construction/initialization from the actions that take place at the start of the round, and have `new_operating_round` and `new_stock_round` call them individually, with a call to `log_new_round` in between, if we really wanted to keep that function around.)

Reply:
why don't you just call log_new_round in the base/round constructor?

Reply:
I've been trained not to call polymorphic functions on an object that hasn't been fully constructed yet. For example if we construct an operating round object and it starts by calling the base round constructor, which then calls `log_new_round`, which is overridden by the operating round, but the operating round method looks at instance variables that haven't been set yet in its own constructor, that would be bad. This can happen very easily in C++; I don't know if maybe it doesn't turn out to be an issue for some reason in Ruby, or if we happen to be able to get away with it in this particular instance. I can take a look.

Reply:
ok, that's fine just delete it from the rounds